### PR TITLE
Windows support

### PR DESCRIPTION
**Pull request by arturkow2000**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -39,3 +45,4 @@ sensors = ["glob"]
 
 [dev-dependencies]
 float-cmp = "0.6.0"
+chrono = "0.4.13"
```
Comment (cjbassi) [inline Cargo.toml:48]: Missing newline.

File: README.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ psutil = { version = "3.1.0", default-features = false, features = ["process"] }
 
 ## Platform Support
 
-Currently, only Linux and macOS are supported, but support is planned for all major platforms.
+Currently, only Linux, macOS and Windows are supported, but support is planned for all major platforms.
```
Comment (cjbassi) [inline README.md:28]: I think this should be changed to: "Currently supported platforms include Linux, macOS, and Windows each with varying levels of implementation."

File: .github/workflows/rust-ci.yml [hunk 3]
```diff
@@ -29,7 +30,12 @@ jobs:
           command: check
 
   test:
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os:
+          - ubuntu-latest
+          - windows-latest
```
Comment (cjbassi) [inline .github/workflows/rust-ci.yml:38]: Should add macOS-latest to the test suite.

File: examples/ps.rs [hunk 4]
```diff
@@ -16,16 +16,19 @@ fn main() {
...
+					"{:>6} {:>2.1} {:>2.1} {}",
+					p.pid(),
+					p.cpu_percent().unwrap_or(0.0f32),
+					p.memory_percent().unwrap_or(0.0f32),
+					p.cmdline()
+						.unwrap_or(None)
+						.unwrap_or_else(|| format!("[{}]", p.name().unwrap())),
+				);
+			}
+			Err(_) => {}
```
Comment (cjbassi) [inline examples/ps.rs:31]: Should probably log this error.

File: src/cpu/sys/windows/cpu_times.rs [hunk 5]
```diff
@@ -0,0 +1,88 @@
+use crate::cpu::CpuTimes;
```
Comment (cjbassi) [inline src/cpu/sys/windows/cpu_times.rs:1]: Imports should be organized into these sections and in this order: std, dependencies, and crate. So these crate dependencies should be moved to line 19. There's a few cases of this across the PR.

File: src/process/os/windows.rs [hunk 6]
```diff
@@ -25,7 +32,63 @@ pub trait ProcessExt {
 
 impl ProcessExt for Process {
 	fn environ(&self) -> ProcessResult<HashMap<String, String>> {
-		todo!()
+		let mut delay = 1;
```
Comment (cjbassi) [inline src/process/os/windows.rs:35]: This section looks like it could use some comments explaining the code including the magic numbers.

**Comment by arturkow2000** [comment]
Commit message contains summary of changes.
Failed checks are unrelated to Windows port - macOS tests are now enabled so errors came out.
This is not ready for merge yet, still few essential things are missing.

**Comment by cjbassi** [comment]
Error handling has been changed from snafu to thiserror so it would be good if you could rebase and change the error handling you added to thiserror. This is looking great so far btw.

**Pull request closed without merging**